src/sys/unix.rs

```diff
@@ -1479,6 +1501,28 @@ impl crate::Socket {
         }
     }
 
+    /// Sets `SO_PEERCRED` to null on the socket.
+    /// This is a Cygwin extension.
+    ///
+    /// Normally the Unix domain sockets of Cygwin are implemented by TCP sockets,
+    /// so it performs a handshake on `connect` and `accept`, to verify the remote
+    /// connection and exchange peer cred info. Therefore, `connect` blocks until
+    /// the server `accept`s it. This option turns off the handshake. Both client
+    /// and server should turn it off to let `connect` returns immediately.
```

Comment:
This text assumes too much about the reader. The reader is not aware of the discussion in this github issue.
```suggestion
    /// connection and exchange peer cred info. At the time of writing, this
    /// means that `connect` on a Unix domain socket will block until the server
    /// calls `accept` on Cygwin. This behavior is inconsistent with most other
    /// platforms, and this option can be used to disable that.
```